pkg/ingress/controller/openstack/octavia.go

```diff
@@ -498,48 +498,42 @@ func (os *OpenStack) CreatePolicyRules(lbID, listenerID, poolID, host, path stri
 
 	log.WithFields(log.Fields{"lb": lbID, "listenerID": listenerID, "policyID": policy.ID}).Info("policy created")
 
-	if host != "" {
-		log.WithFields(log.Fields{"type": l7policies.TypeHostName, "host": host, "policyID": policy.ID, "listenerID": listenerID}).Debug("creating policy rule")
+	var opts l7policies.CreateRuleOpts
+	var fields log.Fields
 
-		// Create HOST_NAME type rule
-		_, err = l7policies.CreateRule(os.octavia, policy.ID, l7policies.CreateRuleOpts{
+	if host != "" {
+		opts = l7policies.CreateRuleOpts{
 			RuleType:    l7policies.TypeHostName,
 			CompareType: l7policies.CompareTypeEqual,
 			Value:       host,
-		}).Extract()
-		if err != nil {
-			return fmt.Errorf("error creating l7 rule: %v", err)
 		}
-
-		_, err = os.waitLoadbalancerActiveProvisioningStatus(lbID)
-		if err != nil {
-			return fmt.Errorf("error waiting for loadbalancer %s to be active: %v", lbID, err)
-		}
-
-		log.WithFields(log.Fields{"type": l7policies.TypeHostName, "host": host, "policyID": policy.ID, "listenerID": listenerID}).Info("policy rule created")
-	}
-
-	if path != "" {
-		log.WithFields(log.Fields{"type": l7policies.TypePath, "path": path, "policyID": policy.ID, "listenerID": listenerID}).Debug("creating policy rule")
-
-		// Create PATH type rule
-		_, err = l7policies.CreateRule(os.octavia, policy.ID, l7policies.CreateRuleOpts{
+		fields = log.Fields{"type": l7policies.TypePath, "host": host, "policyID": policy.ID, "listenerID": listenerID}
+	} else if path != "" {
```

Comment:
the `host` and `path` could be specified at the same time, so should not use `else if` here.